dietpi/dietpi-ddns

```diff
@@ -0,0 +1,372 @@
+			In combination with a custom provider, this is used for HTTP authentication.
+  -t <timespan>		Duration between DDNS updates in minutes (optional, defaults to 10)
+Available providers:
+  <custom>		Full URL to update against a custom DDNS provider
+  DuckDNS		Read more: https://www.duckdns.org/about.jsp
+  No-IP			Read more: https://www.noip.com/about
+'
+
+# Load DietPi-Globals
+. /boot/dietpi/func/dietpi-globals
+readonly G_PROGRAM_NAME='DietPi-DDNS'
+G_CHECK_ROOT_USER
+G_CHECK_ROOTFS_RW
+G_INIT
+
+# Variables
+COMMAND=
+PROVIDER=
+DOMAINS=
+USERNAME=
+PASSWORD=
+TIMESPAN=
+
+# --------------------------------------------
+# Functions
+# --------------------------------------------
+# Process input: Requires script input to be passed
+Input()
+{
+	while [[ $1 ]]
+	do
+		case "$1" in
+			'-d') shift; DOMAINS=$1; shift;;
+			'-u') shift; USERNAME=$1; shift;;
+			'-p') shift; PASSWORD=$1; shift;;
+			'-t') shift; TIMESPAN=$1; shift;;
+			'apply') COMMAND=$1; shift; PROVIDER=$1; shift;;
+			'remove') COMMAND=$1; shift;;
+			*) G_DIETPI-NOTIFY 1 "Invalid input ($1). Aborting...$USAGE"; exit 1;;
+		esac
+	done
+}
+
+# Read current settings from existing cURL command, if not set via input already
+Read()
+{
+	[[ -f '/var/lib/dietpi/dietpi-ddns/update.sh' ]] || return
+	local command=$(tail -1 /var/lib/dietpi/dietpi-ddns/update.sh)
+
+	# DuckDNS
+	if [[ $command == *'duckdns.org'* ]]
+	then
+		[[ $PROVIDER ]] || PROVIDER='DuckDNS'
+		[[ $DOMAINS ]] || DOMAINS=${command#*domains=} DOMAINS=${DOMAINS%&token*}
+		[[ $PASSWORD ]] || PASSWORD=${command#*token=} PASSWORD=${PASSWORD%\'*}
+
+	# No-IP
+	elif [[ $command == *'noip.com'* ]]
+	then
+		[[ $PROVIDER ]] || PROVIDER='No-IP'
+		[[ $DOMAINS ]] || DOMAINS=${command#*hostname=} DOMAINS=${DOMAINS%\'*}
+		[[ $USERNAME ]] || USERNAME=${command#*\'} USERNAME=${USERNAME%%:*}
+		[[ $PASSWORD ]] || PASSWORD=${command#*:} PASSWORD=${PASSWORD%%\'*}
+
+	# Dynu
+	elif [[ $command == *'dynu.com'* ]]
+	then
+		[[ $PROVIDER ]] || PROVIDER='Dynu'
+		[[ $DOMAINS ]] || DOMAINS=${command#*hostname=} DOMAINS=${DOMAINS%&password*}
+		[[ $PASSWORD ]] || PASSWORD=${command#*password=} PASSWORD=${PASSWORD%\'*}
+
+	# Custom
+	else
+		[[ $PROVIDER ]] || PROVIDER=${command%\'*} PROVIDER=${PROVIDER##*\'}
+		# HTTP authentication
+		if [[ $command == *' -u '* ]]
+		then
+			[[ $USERNAME ]] || USERNAME=${command#*\'} USERNAME=${USERNAME%%:*}
+			[[ $PASSWORD ]] || PASSWORD=${command#*:} PASSWORD=${PASSWORD%%\'*}
+		fi
+	fi
+
+	# Time span
+	[[ ! $TIMESPAN && -f '/var/spool/cron/crontabs/dietpi-ddns' ]] || return
+	TIMESPAN=$(tail -1 /var/spool/cron/crontabs/dietpi-ddns)
+	TIMESPAN=${TIMESPAN#\*/} TIMESPAN=${TIMESPAN%% *}
+}
+
+# Apply chosen settings and create a Cron job for dynamic DNS updates
+Apply()
+{
+	# Generate URL
+	local url=$PROVIDER
+	# - DuckDNS
+	if [[ $PROVIDER == 'DuckDNS' ]]
+	then
+		url="https://www.duckdns.org/update?domains=$DOMAINS&token=$PASSWORD"
+		unset -v USERNAME PASSWORD
```

Comment:
Any reason for this line?  Just asking because it makes the 'Token' field disappear.

Reply:
Ah, I didn't think about the values shown in the menu afterwards. Currently this is required to not have HTTP authentication used.

But then we need to a separate variable for this:
```
local http_auth=1
...
if [[ $PROVIDER == 'DuckDNS' ]]
then
	unset -v http_auth
...
...${http_auth:+ -u '$USERNAME:$PASSWORD'}...
```
Will implement this now.

Reply:
Done: https://github.com/MichaIng/DietPi/commit/5ce84c7ed90ffd6c882e18ee2f2f3c2854647838